clients/drcachesim/simulator/caching_device_stats.h

```diff
@@ -115,6 +141,10 @@ class caching_device_stats_t {
     // Print out write invalidations if cache is coherent.
     bool is_coherent_;
 
+    // References to the properties with statistics are held in the map with the
+    // statistic name as the key. Sample map element: {HITS, num_hits_}
+    std::map<metric_name_t, int_least64_t &> stats_map_;
+
```

Comment:
Please add some kind of test.  Maybe in drcachesim_unit_tests or a new similar test?  Xref #4842.

Reply:
I have added unit test in the `drcachesim_unit_tests`. I couldn't test `Invalidates` metric properly because of the bug/issue that I described here: https://github.com/DynamoRIO/dynamorio/issues/5031

Reply:
OK, best to put a `XXX i#5031: ` or `TODO i#5031: ` comment in the test about adding that metric.